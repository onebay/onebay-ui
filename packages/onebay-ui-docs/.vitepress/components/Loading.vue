<template>
  <div class="loading-container">
    <svg
      class="loading"
      t="1605617717566"
      viewBox="0 0 1024 1024"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
      p-id="1397"
      width="200"
      height="200"
    >
      <path
        d="M910.222 455.111V512H682.667v-56.889h227.555z m-625.778 0V512H56.89v-56.889h227.555z m170.667 227.556H512v227.555h-56.889V682.667z m0-625.778H512v227.555h-56.889V56.89z m341.333 699.733l-39.822 39.822-159.289-159.288 39.823-39.823 159.288 159.29zM369.778 329.956l-39.822 39.822-159.29-159.29 39.823-39.821 159.289 159.289z m-39.822 267.377l39.822 39.823-159.29 159.288-39.821-39.822 159.289-159.289z m426.666-426.666l39.822 39.822-159.288 159.289-39.823-39.822 159.29-159.29z"
        fill="#1296db"
        p-id="1398"
        data-spm-anchor-id="a313x.7781069.0.i0"
        class="selected"
      ></path>
    </svg>
  </div>
</template>
<style lang="scss">
.loading-container {
  position: absolute;
  z-index: 99;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.loading {
  width: 3rem;
  height: 3rem;
  animation: myRotate 3s linear infinite;
}
@keyframes myRotate {
  0% {
    -webkit-transform: rotate(0deg);
  }
  50% {
    -webkit-transform: rotate(180deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}
</style>