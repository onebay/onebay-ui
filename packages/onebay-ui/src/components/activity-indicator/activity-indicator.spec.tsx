import { mount } from '@vue/test-utils'
import ActivityIndicator from './index'
import Loading from '../loading/index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(ActivityIndicator, {
    slots,
    components: { Loading },
    props: { ...values }
  })
}

describe('ActivityIndicator Snap', () => {
  it('render initial ActivityIndicator', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props size', () => {
    const wrapper = factory({ size: 32 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props color', () => {
    const wrapper = factory({ color: '#13CE66' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props content', () => {
    const wrapper = factory({ content: '加载中...' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props mode', () => {
    const wrapper = factory({ mode: 'center' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props isOpened', () => {
    const wrapper = factory({ isOpened: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})
