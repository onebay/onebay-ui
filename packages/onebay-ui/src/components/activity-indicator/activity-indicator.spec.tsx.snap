// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator Snap render ActivityIndicator -- props color 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
        style="border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;"
      />
      <div
        class="ob-loading__ring"
        style="border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;"
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
        style="border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props content 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
      />
      <div
        class="ob-loading__ring"
        style=""
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props isOpened 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
      />
      <div
        class="ob-loading__ring"
        style=""
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props mode 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--center ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
      />
      <div
        class="ob-loading__ring"
        style=""
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props size 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style="width: 32px; height: 32px;"
    >
      <div
        class="ob-loading__ring"
        style="width: 32px; height: 32px;"
      />
      <div
        class="ob-loading__ring"
        style="width: 32px; height: 32px;"
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
        style="width: 32px; height: 32px;"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;

exports[`ActivityIndicator Snap render initial ActivityIndicator 1`] = `
<div
  class="ob-activity-indicator ob-activity-indicator--isopened"
>
  <div
    class="ob-activity-indicator__body"
  >
    <div
      class="ob-loading"
      style=""
    >
      <div
        class="ob-loading__ring"
      />
      <div
        class="ob-loading__ring"
        style=""
      />
      <div
        class="ob-loading__ring"
        data-id="[object Object]"
      />
    </div>
  </div>
  <div
    class="ob-activity-indicator__content"
  >
    
    
  </div>
</div>
`;
